server/index.js

```diff
@@ -236,20 +236,29 @@ export default class Server {
 
       // It's very important keep this route's param optional.
       // (but it should support as many as params, seperated by '/')
-      // Othewise this will lead to a pretty simple DOS attack.
+      // Otherwise this will lead to a pretty simple DOS attack.
       // See more: https://github.com/zeit/next.js/issues/2617
       '/_next/:path*': async (req, res, params) => {
         const p = join(__dirname, '..', 'client', ...(params.path || []))
         await this.serveStatic(req, res, p)
       },
 
       // It's very important keep this route's param optional.
-      // (but it should support as many as params, seperated by '/')
-      // Othewise this will lead to a pretty simple DOS attack.
+      // (but it should support as many as params, separated by '/')
+      // Otherwise this will lead to a pretty simple DOS attack.
       // See more: https://github.com/zeit/next.js/issues/2617
       '/static/:path*': async (req, res, params) => {
-        const p = join(this.dir, 'static', ...(params.path || []))
-        await this.serveStatic(req, res, p)
+        const rootPaths = this.nextConfig.rootPaths || ['.']
+        let filePath = null
+        for (let root of rootPaths) {
+          filePath = join(this.dir, root, 'static', ...(params.path || []))
+          try {
+            await access(filePath, (fs.constants || fs).R_OK)
+            break
+          } catch (err) {
```

Comment:
Should it be logged? Or it's ok to leave it silent?

Reply:
Since `fs.exists` is sadly deprecated, this is just being used to see which of the root paths the static file exists in, so it's probably not desirable to log errors encountered while searching. Unfortunately, catching an error is the only way to tell if a file does not exist.